servicelayer/taskqueue.py

```diff
@@ -319,7 +319,6 @@ def mark_for_retry(self, task):
             f" for retry after NACK"
         )
 
-        pipe.sadd(make_key(stage_key, "pending"), task.task_id)
```

Comment:
Removing this since we are calling `add_task` if a task is no longer being tracked by Redis and it is about to be NACK'ed. `add_task` already increases the number of pending tasks, so this is no longer needed.